spec/integration/action_spec.rb

```diff
@@ -1,31 +1,150 @@
 require 'verbalize/action'
 
 describe Verbalize::Action do
-  describe '.verbalize' do
+  let(:a) { double(divide_by: 20) }
+
+  let(:simple_action) do
+    Class.new do
+      include Verbalize::Action
+
+      input :a, :b, :should_fail
+
+      def call
+        fail! 'Are you crazy?!? You can’t divide by zero!' if should_fail
+        a.divide_by(b)
+      end
+    end
+  end
+
+  describe '.call' do
     it 'returns a Verbalize::Failure instance on fail!' do
-      some_class = Class.new do
+      result = simple_action.call(a: a, b: 10, should_fail: true)
+
+      expect(result).to be_an_instance_of(Verbalize::Failure)
+      expect(result).to be_failure
+      expect(result.failure).to eq 'Are you crazy?!? You can’t divide by zero!'
+    end
+
+    it 'returns a Verbalize::Success instance on success' do
+      result = simple_action.call(a: a, b: 10, should_fail: false)
+
+      expect(result).to be_an_instance_of(Verbalize::Success)
+      expect(result).to be_success
+      expect(result.value).to eq 20
+    end
+
+    it 'short circuits the call block on failure' do
+      simple_action.call(a: a, b: 10, should_fail: true)
+
+      expect(a).not_to have_received(:divide_by)
+    end
+
+    it 'stubbed failures return a Verbalize::Falure on `call`' do
+      allow(simple_action).to receive(:perform).and_throw(described_class::THROWN_SYMBOL, 'foo error')
+
+      result = simple_action.call(a: a, b: 10, should_fail: false)
+
+      expect(result).not_to   be_success
+      expect(result).to       be_failed
+      expect(result.failure).to eq 'foo error'
+    end
+
+    it 'fails up multiple levels' do
+      some_inner_inner_class = Class.new do
         include Verbalize::Action
 
         def call
-          fail!('Some error')
+          fail! :some_failure_message
         end
       end
 
-      expect(some_class.call).to be_an_instance_of(Verbalize::Failure)
+      some_inner_class = Class.new do
+        include Verbalize::Action
+      end
+
+      some_inner_class.class_exec(some_inner_inner_class) do |interactor_class|
```

Comment:
Nitpick but I would change interactor_class to action. I guess I should originally used some_inner_action instead of some_inner_class though... (Not a blocker)